resources/sdk/pureclouddotnet/templates/Configuration.mustache

```diff
@@ -62,6 +66,47 @@ namespace {{packageName}}.Client
             Timeout = timeout;
             ShouldRefreshAccessToken = shouldRefreshAccessToken;
             RefreshTokenWaitTime = refreshTokenWaitTime;
+
+            if (logger != null)
+                Logger = logger;
+            else
+                Logger = new Logger();
+
+            if (String.IsNullOrEmpty(configFilePath))
+            {
+                string homePath = (Environment.OSVersion.Platform == PlatformID.Unix || 
+                                Environment.OSVersion.Platform == PlatformID.MacOSX)
+                    ? Environment.GetEnvironmentVariable("HOME")
+                    : Environment.ExpandEnvironmentVariables("%HOMEDRIVE%%HOMEPATH%");
+                ConfigFilePath = Path.Join(homePath, "genesysclouddotnet", "config");
+            }
+            else
+            {
+                ConfigFilePath = configFilePath;
+            }
+
+            // applyConfigFromFile();
+            if (AutoReloadConfig)
+            {
+                ThreadStart childref = new ThreadStart(RunConfigChecker);
+                Console.WriteLine("In Main: Creating the Child thread");
+                Thread childThread = new Thread(childref);
+                childThread.Start();
+            }
+        }
+
+        private void applyConfigFromFile()
+        {
+
+        }
+
+        public static void RunConfigChecker()
+        {
+            while (true)
+            {
+                Console.WriteLine("runConfigChecker");
+                Thread.Sleep(2000);
```

Comment:
Just to double check...... I am assuming this thread. Sleep was running off the thread you spun off for checking the config.  Also, don't hard code the value using a magic number.  Put 2000 in a constant at the top of the file so we know exactly why this parameter is there.